### PR TITLE
[MRG] Connect structure Update API

### LINKED ISSUE #209: Connect Model Structure Update API
Connect model structure update API, enable users to select a layer, modify layer parameters and submit the modifications.

### PR DESCRIPTION
**Pull request by univerone**
Closes #209

#### What has been done to verify that this works as intended?

Firstly, click the finetune button
Screenshot_2021-02-03 ML ModelCI(7)

Then click on the model structure graph, edit relevant parameters
Screenshot_2021-02-03 ML ModelCI(6)

Submit the modification
Screenshot_2021-02-03 ML ModelCI(3)

The modified model structure is registered in the ModelDB
Screenshot_2021-02-03 ML ModelCI(8)

#### Before submitting this PR, please make sure you have:

- [ ] run `python -m pytest tests/` and confirmed all checks still pass.
- [ ] verified that any code or assets from external sources are properly credited.

### CODE REVIEW

File: frontend/src/config.ts [hunk 1]
```diff
@@ -1,6 +1,9 @@
 export default {
-    default: {
-        modelURL: 'http://localhost:8000/api/v1/model',
-        visualizerURL: 'http://localhost:8000/api/v1/visualizer'
-    }
-  }
\ No newline at end of file
+  default: {
+    modelURL: 'http://localhost:8000/api/v1/model',
```
Comment (YuanmingLeee) [inline frontend/src/config.ts:3]: you can set the host address as a variable and use it to config these urls:
```js
HOST_URL = 'http://localhost:8000';

modelURL = `${HOST_URL}/api/v1/model`;
```

File: modelci/experimental/model/model_structure.py [hunk 2]
```diff
@@ -40,6 +42,10 @@ class LayerType(Enum):
     TANH = 'torch.nn.Tanh'
     BN_1D = 'torch.nn.BatchNorm1d'
     BN_2D = 'torch.nn.BatchNorm2d'
+    MP_1D = 'torch.nn.MaxPool1d'
```
Comment (YuanmingLeee) [inline modelci/experimental/model/model_structure.py:45]: Currently, we do not need this layer.

File: modelci/experimental/model/model_structure.py [hunk 3]
```diff
@@ -174,7 +174,37 @@ class BatchNorm2d(_BatchNorm):
     __required_type__ = LayerType.BN_2D
 
 
-_LayerType = Union[LinearLayer, Conv1d, Conv2d, ReLU, Tanh, BatchNorm1d, BatchNorm2d]
+class _MaxPool(ModelLayer, abc.ABC):
```
Comment (YuanmingLeee) [inline modelci/experimental/model/model_structure.py:177]: don't need these currently

File: modelci/experimental/model/model_structure.py [hunk 4]
```diff
@@ -222,3 +252,36 @@ class Structure(BaseModel):
         default_factory=dict,
         example={'conv1': {'fc1': 'A'}}
     )
+
+    @classmethod
+    def from_model(cls, model): #noqa
```
Comment (YuanmingLeee) [inline modelci/experimental/model/model_structure.py:257]: Your parser will encounter missing of layers such as 'ReLU', 'Dropout' that do not have weight parameters. There is a better implementation idea for this. Do not care about how we get the model structure from a model weight file now. 

Torchviz is good enough for us.

**Comment by HuaizhengZhang** [comment]
image
The property should be very closed to the corresponding layer

File: frontend/src/pages/Visualizer/utils/mock.tsx [hunk 5]
```diff
@@ -0,0 +1,1918 @@
+const mockStructure = {
```
Comment (YuanmingLeee) [inline frontend/src/pages/Visualizer/utils/mock.tsx:1]: You should use JSON to pass the mocked data, and use `require('xxx..json')` to load the JSON file. Data should not be kept in typescript.

**Review by YuanmingLeee — APPROVED** [review]
